core/server/worker/src/main/java/alluxio/worker/block/BlockWorker.java

```diff
@@ -438,20 +438,30 @@ void closeUfsBlock(long sessionId, long blockId)
    *
    * @param request the block read request
    * @return a block reader to read data from
+   * @throws BlockAlreadyExistsException if it fails to commit the block to Alluxio block store
+   *         because the block exists in the Alluxio block store after opening the ufs block reader
+   * @throws BlockDoesNotExistException if the requested block does not exist in this worker
+   * @throws InvalidWorkerStateException if blockId does not belong to sessionId
+   * @throws WorkerOutOfSpaceException if there is no enough space
+   * @throws IOException if it fails to get block reader
    */
-  BlockReader getBlockReader(BlockReadRequest request) throws IOException,
-      BlockDoesNotExistException, InvalidWorkerStateException,
-      BlockAlreadyExistsException, WorkerOutOfSpaceException;
+  BlockReader getBlockReader(BlockReadRequest request) throws
```

Comment:
shall this `BlockWorker` interface to extend `WorkerInternalBlockWorker`?

Reply:
Removed the WorkerInternalBlockWorkerImpl and let the BlockWorker interface directly extend the WorkerInternalBlockWorker